### DISCUSSION TITLE
useConvexPolyhedron not working?

### DISCUSSION THREAD

**Discussion by danielacorner** [pmndrs/use-cannon#204] [Q&A] [answered]
since migrating from `react-three-fiber` to `@react-three/fiber` I can't get `useConvexPolyhedron` from `@react-three/cannon` working again

sandbox: https://codesandbox.io/s/useconvexpolyhedron-bug-bqdqz?file=/src/App.js

```tsx
function Shape() {
  // ? doesn't work with useConvexPolyhedron
  // const polyRef = useConvexPolyhedron(() => ({
  //   mass: 1,
  //   args: new THREE.IcosahedronGeometry(1, 1)
  // }));
  // works with useBox
  const boxRef = useBox(() => ({
    mass: 1,
    args: [1,1,1]
  }));
  return <mesh ref={boxRef}>
    <boxBufferGeometry/>
    <meshBasicMaterial/>
  </mesh>;
}
```

when I uncomment `useConvexPolyhedron`:

![image](https://user-images.githubusercontent.com/9141196/121764671-b2d43180-cb13-11eb-846e-f2aa5b7950c1.png)

any ideas? thanks 🙏

**Comment by stockhuman** [comment, marked as answer]
Hi Daniel,

The ability to pass `THREE. Geometry` directly is no longer supported as of [0.6](https://github.com/pmndrs/use-cannon/releases/tag/v0.6.0), check out the convex polyhedron example to see how it's used.

If you dig into the source, you'll see that use-cannon is trying to `map` vertex arrays, but none were supplied.

Good luck!

**Reply by drcmda** [reply]
https://codesandbox.io/s/physics-with-convex-polyhedrons-08s1u

**Reply by danielacorner** [reply]
perfect, thanks!
